[INTERNAL] CustomTask: Enhance implementation documentation (#581)
SAP/ui5-tooling: docs/pages/extensibility/CustomTasks.md

 * @param {object} parameters.taskUtil Specification Version dependent interface to a
 *                [TaskUtil]{@link module:@ui5/builder.tasks.TaskUtil} instance
 * @param {object} parameters.options Options
 * @param {string} parameters.options.projectName Project name
 * @param {string} [parameters.options.projectNamespace] Project namespace
 * @param {string} [parameters.options.configuration] Task configuration if given in ui5.yaml
 * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written or rejecting in case of an error
 */
module.exports = async function({workspace, dependencies, taskUtil, options}) {
	// [...]
};
````

**Parameters:**

- **`workspace`**: A [DuplexCollection](https://sap.github.io/ui5-tooling/api/module-@ui5_fs.DuplexCollection.html) to read and write [Resources](https://sap.github.io/ui5-tooling/api/module-@ui5_fs.Resource.html) for the project that is currently being built
- **`dependencies`**: A [ReaderCollection](https://sap.github.io/ui5-tooling/api/module-@ui5_fs.ReaderCollection.html) to read [Resources](https://sap.github.io/ui5-tooling/api/module-@ui5_fs.Resource.html) of the project's dependencies

> - **`taskUtil`**: See [details below](http://localhost:8000/pages/extensibility/CustomTasks/#helper-class-taskutil)

Reviewer: ```suggestion
- **`taskUtil`**: See [details below](https://sap.github.io/ui5-tooling/pages/extensibility/CustomTasks/#helper-class-taskutil)
```

- **`options.projectName`**: The name of the project currently being built. *Example: `my.library`*
- **`options.projectNamespace`**: The namespace of the project. *Example: `my/library`*
- **`options.configuration`**: The task configuration as defined in the project's ui5.yaml. See [Configuration](#Configuration)

**Returns:**

A Promise that resolves once the task has completed and all new or modified resources have been written to the workspace.

In case of errors the promise should reject with an [Error object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error), causing the build to abort. 

!!! warning
    Depending on your project setup, UI5 Tooling tends to open many files simultaneously during a build. To prevent errors like `EMFILE: too many open files`, we urge custom task implementations to use the [graceful-fs](https://github.com/isaacs/node-graceful-fs#readme) module as a drop-in replacement for the native `fs` module.

### Example: lib/tasks/generateMarkdownFiles.js

> The following code snippets shows an example how a task implementation could look like. This task uses a generic "toMarkdown" library to transform text to [markdown](https://daringfireball.net/projects/markdown/) and writes out the newly created markdown files.

Reviewer 2: ```suggestion
The following code snippet shows an example of what a task implementation might look like. This task uses a generic "toMarkdown" library to transform text to [markdown](https://daringfireball.net/projects/markdown/) and writes out the newly created markdown files.
```
